pkg/provider/vault/vault.go

```diff
@@ -219,6 +219,9 @@ func (v *client) readSecret(ctx context.Context, path, version string) (map[stri
 
 	byteMap := make(map[string][]byte, len(secretData))
 	for k, v := range secretData {
+		if v == nil {
+			continue
+		}
 		switch t := v.(type) {
 		case string:
```

Comment:
This fixes the error but the `secretData` map which we return does not have the nil value in it. I think we should check the nil value in the type assertion below and set the map key to the nil value `byteMap[k] = []byte(nil)`.

This way the `byteMap` contains the very same data as in vault. 

Could you please add a test to verify that behavior?

Reply:
Makes sense! Updated ✅ 

I've also added a couple of tests not closely related to the fix for testing the func `GetSecretMap` properly. I hope it's ok,  let me know otherwise.